src/app/services/xml-parsers/apparatus-entries-parser.service.ts

```diff
@@ -0,0 +1,128 @@
+    shareReplay(1),
+  );
+
+  private appEntryTagName = 'app';
+  private lemmaTagName = 'lem';
+  private readingTagName = 'rdg';
+  private readingGroupTagName = 'rdgGrp';
+  private noteTagName = 'note';
+  private significantRdgNumber = 0;
+  private parsedWitnesses = {};
+
+  constructor(
+    private editionDataService: EditionDataService,
+    private genericParserService: GenericParserService,
+    private witParserService: WitnessesParserService,
+  ) {
+  }
+
+  private parseAppEntriesList(document: XMLElement): ApparatusEntriesList {
+    const appEntries = Array.from(document.querySelectorAll<XMLElement>(this.appEntryTagName));
+
+    return this.parseAppEntries(appEntries);
+  }
+
+  private parseAppEntries(appEntries: XMLElement[]) {
+    this.witParserService.witnesses$.subscribe(data => this.parsedWitnesses = data);
+    const parsedEntries = appEntries.map((appEntry) => this.parseAppEntry(appEntry));
+
+    return arrayToMap(parsedEntries, 'id');
+  }
+
+  private parseAppEntry(appEntry: XMLElement): ApparatusEntry {
+    this.significantRdgNumber = 0;
+
+    return {
+      type: 'ApparatusEntryComponent',
+      id: appEntry.getAttribute('xml:id') || xpath(appEntry),
+      attributes: this.genericParserService.parseAttributes(appEntry),
+      content: this.parseAppReadings(appEntry),
+      note: this.genericParserService.parse(appEntry.querySelector<XMLElement>(this.noteTagName)),
+      variance: this.getVariance(),
+      originalEncoding: getOuterHTML(appEntry),
+    };
+  }
+
+  private parseAppReadings(appEntry: XMLElement): ApparatusReading[] {
+    return Array.from(appEntry.querySelectorAll(`${this.lemmaTagName}, ${this.readingTagName}`))
+      .map((child: XMLElement) => this.parseAppReading(child));
+  }
+
+  private parseAppReading(rdg: XMLElement): ApparatusReading {
+    return {
+      type: 'ApparatusReadingComponent',
+      tagname: rdg.tagName,
+      id: rdg.getAttribute('xml:id') || xpath(rdg),
+      attributes: this.genericParserService.parseAttributes(rdg),
+      wits: this.parseReadingWitnesses(rdg) || [],
+      content: this.parseAppReadingContent(rdg),
+      significant: this.parseReadingSignificance(rdg),
+    };
+  }
+
+  private parseReadingWitnesses(rdg: XMLElement) {
+    return rdg.getAttribute('wit')?.split('#')
+      .map((el) => removeSpaces(el))
+      .filter((el) => el.length !== 0);
+  }
+
+  private parseAppReadingContent(rdg: XMLElement) {
+    return Array.from(rdg.childNodes)
+      .map((child: XMLElement) => {
+        if (child.nodeName === this.appEntryTagName) {
+          return {
+            type: 'ApparatusEntryComponent',
+            id: child.getAttribute('xml:id') || xpath(child),
+            attributes: {},
+            content: [],
+          };
+        }
+
+        return this.genericParserService.parse(child);
+      });
+  }
+
+  private parseReadingSignificance(rdg: XMLElement) {
+    return rdg.parentElement.tagName === this.readingGroupTagName ?
```

Comment:
What if you have consider not significant those readings with `@type="orthographic"` and you have this encoding:
```
<rdgGrp resp="m">
   <rdg type="orthographic">Loremm</rdg>
   <rdg>Irem</rdg>
</rdgGrp>
```
in this case, when parsing the first `rdg`, since parent element is a `<rdgGrp>`, it will run the significance check on parent element attributes only and the reading will be marked as significant. You should probably also check whether the `<rdg>` as attributes or not, and if it has, check those first (then check parent's if the reading results to be significant).

Reply:
Done